src/api.js

```diff
@@ -23,14 +23,16 @@ const API = {
       'onRequestCompleted',
       'requestData',
       'requestHeaders',
-      'urlRoot'
+      'urlRoot',
+      'superagent'
     ]))
   },
-	
+
 	request(options = {}) {
 
 		let { method, data, endpoint, onSuccess, onError, fileData } = options;
     let requestData, requestHeaders, doRequest;
+    const request = this.superagent || superagent;
```

Comment:
I am checking for  superagent object in the request options first, then in config and if none found, use the superagent import.